docs/proposals/interceptors.md

```diff
@@ -0,0 +1,196 @@
+# Proposal: Introduce GraphQL Interceptors
+
+_Owners_: @ThisaruGuruge @DimuthuMadushan     
+_Reviewers_: @shafreenAnfar @ThisaruGuruge       
+_Created_: 2022/06/07   
+_Updated_: 2021/07/18     
+_Issue_: [#2001](https://github.com/ballerina-platform/ballerina-standard-library/issues/2001)
+
+
+## Summary
+Introduce the GraphQL Interceptors to enhance the execution flow of the GraphQL services. The GraphQL interceptors can execute custom code before and after the resolver function gets invoked which helps with common stuff like authentication, authorization, logging, observability, etc.
+
+## Goals
+* Introduce service level interceptors
+* Introduce resolver level interceptors
```

Comment:
This isn't a goal of this proposal, right?

Reply:
removed!